spec/helpers/address_helper_spec.rb

```diff
@@ -144,24 +144,21 @@
       expect(helper.sanitize_address(address)).to eq array
     end
 
-    it "returns a valid array if part of the address is nil" do
+    it "returns an array of strings if the address contains non-string values" do
       address = {
-        one: "one two three 1234567890",
-        two: nil,
+        string: "one",
```

Comment:
Would it help the test if the bad values were put into keys that the helper is expecting like `uprn` or `building_and_street_line_1`, rather than `symbol:`?

```suggestion
        building_and_street_line_1: "one",
        building_and_street_line_2: 100,
        town_city: 1.34,
        county: Object.new,
        uprn: Struct.new("Ignored"),
        array: [],
        postcode: "AA1A 1AA",
```

Reply:
Can do, but there are some reasons why I don't feel this is necessary or useful:

* It is the `sanitize_address` method that was throwing the `NoMethodError`.  This method doesn't care about the format of the address, but it does remove UPRN, country and town_city. 

* There are only 5 fields in the support_address - and 3 of those are removed by the `sanitize_method`, so I'd need a number tests to test all types of bad value.

Happy to add if you feel testing the `sanitize_method` needs these specific fields.

Reply:
Thanks for explaining. This makes sense :shipit: